### PR TITLE
solution

### PR DESCRIPTION
**Pull request by ihorandrianov**

### CODE REVIEW

**Review by korniienkokarina — COMMENTED** [review]
good job! 👏

File: src/transformState.js [hunk 1]
```diff
@@ -1,11 +1,63 @@
...
+          return p;
+        }, _state);
+
+        return updatedState;
+      case 'clear':
+        return {};
+      default:
+        break;
+    }
+  };
```
Comment (Denchepornyuk) [inline src/transformState.js:55]: remove comments

File: src/transformState.js [hunk 2]
```diff
@@ -1,11 +1,63 @@
+/* eslint-disable spaced-comment */
 'use strict';
 
 /**
  * @param {Object} state
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const reducer = (_state, action) => {
```
Comment (Denchepornyuk) [inline src/transformState.js:9]: why do you use `_` for `state`?
It's a normal argument for the faction, don't use `_`

File: src/transformState.js [hunk 3]
```diff
@@ -1,11 +1,63 @@
...
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete _state[key];
+        }
+        break;
+      case 'clear':
+        const objectProps = Object.getOwnPropertyNames(_state);
+
+        objectProps.forEach((key) => delete _state[key]);
+        break;
```
Comment (Denchepornyuk) [inline src/transformState.js:25]: you can use `for in` instead of getOwnPropertyNames and forEach

**Review by denys-cheporniuk — CHANGES_REQUESTED** [review]

File: src/transformState.js (hunk 2 above)
Comment (ihorandrianov) [reply src/transformState.js:9]: @Denchepornyuk Its a "no-shadowing" rule in ESlinter

File: src/transformState.js [hunk 4]
```diff
@@ -1,11 +1,36 @@
...
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const reducer = (reducerState, action) => {
+    switch (action.type) {
+      case 'addProperties':
+        for (const key in action.extraData) {
+          reducerState[key] = action.extraData[key];
+        }
```
Comment (Denchepornyuk) [inline src/transformState.js:14]: use Object.assign for adding new values to the object

**Review by denys-cheporniuk — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 5]
```diff
@@ -1,11 +1,34 @@
...
+        break;
+      case 'clear':
+        for (const param in reducerState) {
+          delete reducerState[param];
+        }
+        break;
+      default:
+        break;
+    }
+  };
```
Comment (OleksiiBlezniuk) [inline src/transformState.js:27]: This helper function also can be moved outside `transformState` function in order to make it reusable

**Review by OleksiiBlezniuk — APPROVED** [review]
Good job!

**Pull request closed without merging**